### ISSUE TITLE
Invalid result of running U2Net on Vulkan

### ISSUE THREAD

**Issue by MaksimDanilov** [pytorch/executorch#15490]
### 🐛 Describe the bug

I try to get rid of multiple nn formats (CoreML and Tensorflow-Lite) for U2Net. Exported program for CoreML and XNNPack gives correct result, but If I partition with Vulkan it gives corrupted image after processing. Do I need to test Vulkan only on android devices? How can I understand what happened?

### Versions

Build from source exectorch v1.0.0 on Apple MacBook Pro M1.

cc @SS-JIA @manuelcandales @digantdesai @cbilgin

**Comment by MaksimDanilov** [comment]
Vulkan
<img width="1164" height="346" alt="Image" src="https://github.com/user-attachments/assets/c82d95eb-430f-456c-b648-32a12e317f6c" /> 

XNNPack
<img width="1168" height="349" alt="Image" src="https://github.com/user-attachments/assets/a04131bf-2a56-4d59-8f4f-88c2466a0fb8" />

**Comment by MaksimDanilov** [comment]
Also run this model on Android (build from source executorch with Vulkan and XNNPack enabled and ship as static library to my library) and the same result. GPU seems not invoking on Pixel 9 Pro 'cause neither profiler gives stats on GPU, nor inference time (XNNPack vs Vulkan) doesn't change. On Mac Vulkan do run on GPU.

**Comment by SS-JIA** [comment]
@MaksimDanilov would you be able to provide an export script + model run instructions for your model? I will try to reproduce and figure out the issue!

**Comment by MaksimDanilov** [comment]
I will try to make projects to reproduce the problem to this issue :-)

**Comment by MaksimDanilov** [comment]
https://github.com/MaksimDanilov/executorch_vulkan_problem - done :-)

**Comment by MaksimDanilov** [comment]
@SS-JIA just in case comment above is not get lost

**Comment by SS-JIA** [comment]
@MaksimDanilov sorry, just seeing this - will take a look and keep you updated!

**Comment by SS-JIA** [comment]
@MaksimDanilov thanks for putting up the repo. I was able to reproduce the issue - however, strangely I found that the issue was not in the Vulkan backend.

In your app code, I found that if I just convert the input image to float and normalize it to the 0, 1 range, the model seems to work fine.

```c++
    cv::Mat image_orig = cv::imread(if_, cv::IMREAD_COLOR);
    if (image_orig.empty())
        return -1;

    cv::Mat image;
    image_orig.convertTo(image, CV_32F, 1.0 / 255.0); // Convert
```

For example, I see this result for the horse image:

![Image](https://github.com/user-attachments/assets/7d55d64b-dcf3-4517-a41d-011052a25ae9)

I created a fork of your repo with my changes: https://github.com/SS-JIA/executorch_vulkan_problem

I added a section in the README with the full instructions for how to repro.

Please try it out and lmk if you are still having issues!

**Comment by MaksimDanilov** [comment]
@SS-JIA Hi. Yeap. I know that model should be converted to that range, however forgot about it. I'll come back later with reply. Thank you

**Comment by MaksimDanilov** [comment]
Surprisingly, with xnnpack and coreml this code works.
